src/main/java/com/dyescape/dyescapebot/entity/discord/ActivePunishment.java

```diff
@@ -0,0 +1,133 @@
+package com.dyescape.dyescapebot.entity.discord;
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.time.Instant;
+
+@Entity
+@IdClass(ActivePunishment.PunishmentKey.class)
+@Table(name = "active_punishments")
+public class ActivePunishment implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    // -------------------------------------------- //
+    // COLUMNS
+    // -------------------------------------------- //
+
+    @Id
+    @Column(name = "server_id", nullable = false)
+    private long server;
+
+    @Id
+    @Column(name = "user_id", nullable = false)
+    private long user;
+
+    @Id
+    @Column(name = "type", nullable = false)
+    private String type;
```

Comment:
You sure you want all these columns being primary keys?

> Annotation Type Id:
> Specifies the primary key of an entity. The field or property to which the Id annotation is applied should be one of the following types: any Java primitive type; any primitive wrapper type; String; java.util.Date; java.sql.Date; java.math.BigDecimal; java.math.BigInteger.

Reply:
Yes. The bot is open source, and with some future plans, I want this bot to be fully public to the point where people can invite it to their own server. This means that an active punishment must be identified by server, user and punishment type. That way, any duplication logic is prevented on database level, which should keep the data as correct as possible. It is possible for the same player to be in different servers and have different punishments on those servers, hence why the three as a combination are a key. That's all done on purpose.

Reply:
So it is impossible to be punished multiple times on a server, in a channel ... with the same type, but different end times?

(If thats impossible, feel free to resolve this)